src/components/SignUp.js

```diff
@@ -1,40 +1,102 @@
+            }
         }
         this.switchToSignUp = this.switchToSignUp.bind(this)
         this.handleChange = this.handleChange.bind(this)
         this.handleSubmit = this.handleSubmit.bind(this)
+        this.passwordChecker = this.passwordChecker.bind(this)
+        this.messageDisplayer = this.messageDisplayer.bind(this)
     }
 
-    switchToSignUp(){
+    switchToSignUp() {
         this.props.childSwitcher()
     }
-    
-    handleChange(event){
-        switch (event.target.type){
+
+    handleChange(event) {
+        let copiedTempState = { ...this.state }
+
+        switch (event.target.name) {
             case "email":
-                this.setState({email: event.target.value.toLowerCase()})
-                break  
-            case "password":
-                this.setState({password: event.target.value})        
+                copiedTempState.tempData.email = event.target.value
+                break
+            case "pass1":
+                copiedTempState.tempData.pass1 = event.target.value
                 break
+            case "pass2":
+                copiedTempState.tempData.pass2 = event.target.value
+                break
+            case "agreed":
+                copiedTempState.tempData.agreed = !copiedTempState.tempData.agreed    
             default:
-                console.log("ohh")          
+                console.log("ohh switch")
         }
+
+        this.setState(copiedTempState)
+        console.log(this.state)
     }
 
-    handleSubmit(event){
-        this.props.signYouUp(this.state)
+    handleSubmit(event) {
+        let copiedTempState = { ...this.state }
+        if(!copiedTempState.tempData.agreed){
+            this.messageDisplayer("You haven't agreed to whatever!")
+        }
+        if(this.passwordChecker(copiedTempState.tempData.pass1, copiedTempState.tempData.pass2)
+            &&
+            copiedTempState.tempData.agreed){
+                copiedTempState.finalData.password = copiedTempState.tempData.pass1
+                copiedTempState.finalData.email = copiedTempState.tempData.email
+                this.setState(copiedTempState)
+                this.props.signYouUp(this.state.finalData)
+        }
+        
+        console.log(this.state)        
         event.preventDefault()
     }
 
+    passwordChecker(pass1, pass2) {
+        const regex = new RegExp("^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$")
+        if(regex.test(pass1)){
+            if(pass1 === pass2){
+                console.log("good passwords")
+                return true
+            }else{
+                this.messageDisplayer("your passwords don't match!")
+                return false
+            }
+                        
+        }else{
+            this.messageDisplayer("Your password is not strong enough")
+            return false
+        }
+        
+        
+    }
+
+    messageDisplayer(message) {
+            let copiedTempState = { ...this.state }
```

Comment:
I suppose the original solution is less complicated, because if I sent it as param, I would need to copy tempState each time I call messageDisplayer such as on lines 77 and 81. But maybe I am not right.

Reply:
Yes, I agree with you. Leave it as it is.

Reply:
Ok, thanks